### PR TITLE
Extend contribution notes, add submodule information, change syntax and fix typos

### PR DESCRIPTION
**Pull request by Lupphes**
I created this pull request to add contribution notes to the documentation. While writing the changes, I also added the required information about the common repository #12 and fixed some typos and updated the markdown syntax.

### CODE REVIEW

**Comment by Lupphes** [comment]
Relates #12

File: README.md [hunk 1]
```diff
@@ -1,58 +1,52 @@
-Welcome to the SCL.org Space
-============================
-This is basically an overview around the sclorg organization, aka what all one can find here.
+# Welcome to the SCL.org Space
 
-Why SCL.org? Since all the repositories, under this organization are somehow connected to the Software Collections concept. Software Collections give you the power to build, install, and use multiple versions of software on the same system, without affecting system-wide installed packages.
+This is an overview of the sclorg organization, aka all one can find here.
```
Comment (phracek) [inline README.md]: Please do not use 'aka'. Use the full names.

File: README.md [hunk 2]
```diff
@@ -1,58 +1,52 @@
...
 
-And since we have had set of new web development tools available as RPMs, tested and available in the directory on [softwarecollections.org](https://www.softwarecollections.org/en/), it is a good idea to re-use those tested packages and make container images from them, that are working fine in OpenShift.
```
Comment (phracek) [inline README.md:7]: Why did you delete the reference? It is not valid?

**Review by phracek — CHANGES_REQUESTED** [review]
Please address my changes if they are feasible.

File: README.md (hunk 2 above)
Comment (Lupphes) [reply README.md:7]: It is still there on line 7. Git diff moved it to the wrong line (under removed Repositories Overview).

File: contribution.md [hunk 3]
```diff
@@ -81,11 +82,25 @@ In the created repositories add also robot account which is mandatory for publis
...
+
+You should be informed about each repository package's requirements, as they can slightly differ. Usually, those packages are needed:
+
+- [distgen](https://github.com/devexp-db/distgen/) - is used for generating image source files,
+- [go-md2ma](https://github.com/cpuguy83/go-md2man) - is used for generating documentation from README.md
+
+1. Fork the repository
+2. Run `git submodule update --init` to download the `common` submodule with `common/common.mk` makefile. We use this submodule for common tests and functions across all our repositories.
+3. Add a feature or bug fix in the templates (src directory) or values (specs/multispec.yml file)
+   1. If you do not change anything in those directories, you do not need to regenerate files
```
Comment (pkubatrh) [inline contribution.md]: This step is not typical at all - we only use distgen generation in a small number of repositories. I would just go with a generic "Add a feature or bug fix" step. If you want to, you can explicitly state that for distgen-enabled repositories the sources are in the `/src` directory and so on.

Comment (Lupphes) [reply contribution.md]: Fixed in the last commit. Mentioned with `*`.

File: README.md (hunk 1 above)
Comment (Lupphes) [reply README.md]: I rewrote some sentences to make them sound more professional :smile:

File: README.md [hunk 4]
```diff
@@ -1,58 +1,53 @@
-Welcome to the SCL.org Space
-============================
-This is basically an overview around the sclorg organization, aka what all one can find here.
+# Welcome to the SCL.org Space
 
-Why SCL.org? Since all the repositories, under this organization are somehow connected to the Software Collections concept. Software Collections give you the power to build, install, and use multiple versions of software on the same system, without affecting system-wide installed packages.
+This document offers a comprehensive introduction to the sclorg organization, highlighting the repositories under our umbrella and providing examples of their practical applications, along with links to relevant external resources.
```
Comment (zmiklank) [inline README.md]:
```suggestion
This document offers a comprehensive introduction to the Software Collections Organization (SCLOrg), highlighting the repositories under our umbrella and providing examples of their practical applications, along with links to relevant external resources.
```

File: README.md [hunk 5]
```diff
@@ -1,58 +1,53 @@
...
 
-And since we have had set of new web development tools available as RPMs, tested and available in the directory on [softwarecollections.org](https://www.softwarecollections.org/en/), it is a good idea to re-use those tested packages and make container images from them, that are working fine in OpenShift.
+The primary focus of SCL.org revolves around the Software Collections concept. Software Collections offer a streamlined approach to building, installing, and utilizing multiple software versions on a single system, without disrupting pre-existing system-wide package installation.
 
-Repositories Overview
----------------------
+In light of recent developments, we have introduced a series of RPMs web development tools. These tools have been thoroughly tested and are accessible via the softwarecollections.org directory. To maximize the utility of these verified packages, we encourage the creation of container images compatible with OpenShift.
```
Comment (zmiklank) [inline README.md]: I have no information about publishing our work on softwarecollections.org. Maybe this was in the history and works no longer for us?
@pkubatrh do you know something more about this, please?

File: README.md [hunk 6]
```diff
@@ -1,58 +1,53 @@
...
-* [centos-release package](https://github.com/sclorg/centos-release-scl): Small RPM package with YUM repository that provides upstream Software Collections built in [CBS](http://cbs.centos.org/) by community (not those that are part of RHSCL).
-* [SCLo CI Tests](https://github.com/sclorg/sclo-ci-tests): Tests that are run in [CentOS CI](http://ci.centos.org) to verify the RPM packages were (re)build fine in [CBS](http://cbs.centos.org/).
-* [RPM List Builder](https://github.com/sclorg/rpm-list-builder): Tool that should help rebuilding SRPMs in [CBS](http://cbs.centos.org/), that were previously made available in different repository (for example RHSCL, but not only this one).
-* [RHSCL Container CI](https://github.com/sclorg/rhscl-container-ci): Scripts that help testing containers on RHEL platform.
+## Repository Overview
 
+- [Container Contribution Guidelines](contribution.md): Learn about what principles to follow when contributing to the sclorg organization, particularly in relation to container repositories.
```
Comment (zmiklank) [inline README.md]:
```suggestion
- [Container Contribution Guidelines](contribution.md): Learn about what principles to follow when contributing to the SCLOrg, particularly in relation to container repositories.
```
AFAIK, in general we use capital letters like this for SCLOrg. This applies also for other occurrences of 'sclorg' in this text.

File: README.md [hunk 7]
```diff
@@ -1,58 +1,53 @@
...
+- [SCLo CI Tests](https://github.com/sclorg/sclo-ci-tests): Tests executed in [CentOS CI](http://ci.centos.org) to verify the successful (re)building of RPM packages in [CBS](http://cbs.centos.org/).
+- [RPM List Builder](https://github.com/sclorg/rpm-list-builder): A tool designed to facilitate the rebuilding of SRPMs in [CBS](http://cbs.centos.org/) that were previously available in different repositories (e.g., RHSCL and others).
+- [RHSCL Container CI](https://github.com/sclorg/rhscl-container-ci): Scripts for testing containers on the RHEL platform.
 
- * https://developer.fedoraproject.org/tools/docker/about.html
- * https://fedoramagazine.org/quick-containers-with-fedora-dockerfiles/
- * https://www.liquidweb.com/kb/how-to-list-and-attach-to-docker-containers/
- * https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux_atomic_host/7/html/overview_of_containers_in_red_hat_systems/introduction_to_linux_containers
+# Where to Learn More About Containers
```
Comment (zmiklank) [inline README.md]: In my opinion is redundant to state such a generic information in SCLOrg's readme.

File: README.md [hunk 8]
```diff
@@ -1,58 +1,53 @@
...
-Reproducible builds are created via Dockerfiles, learn more about Dockerfile basics in the following articles:
+- https://developer.fedoraproject.org/tools/docker/about.html
+- https://fedoramagazine.org/quick-containers-with-fedora-dockerfiles/
+- https://www.liquidweb.com/kb/how-to-list-and-attach-to-docker-containers/
+- https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux_atomic_host/7/html/overview_of_containers_in_red_hat_systems/introduction_to_linux_containers
 
- * https://docs.docker.com/engine/reference/builder/
- * https://developers.redhat.com/blog/2016/02/24/10-things-to-avoid-in-docker-containers/
- * http://docs.projectatomic.io/container-best-practices/
+## How to write a Dockerfile
```
Comment (zmiklank) [inline README.md]: The same for writing dockerfile - this is imo not information, that should be stated in SCLOrg's readme, as is not really related to SCLOrg stuff.

File: contribution.md [hunk 9]
```diff
@@ -1,117 +1,135 @@
...
-This document describes rules and best practices for contributing to http://github.com/sclorg/.
+# Contributing to Container Images
 
+This document outlines the guidelines and best practices for contributing to the container images within the [sclorg organization](http://github.com/sclorg/) on GitHub.
 
 ## Pull Requests
 
-All changes done in the images go through pull request (PR) and require review from another maintainer before merging. A reasonably small change that is extra-obvious and cannot cause any issue can be merged/committed without a review (typically typos in documentation).
-
+All modifications to the repositories and images must be submitted via a pull request (PR) and undergo review by another maintainer before merging. Exceptionally, small and obvious changes that pose no risk (such as fixing typos in documentation) may be merged or committed without review.
```
Comment (zmiklank) [inline contribution.md]:
```suggestion
All modifications to the repositories and images must be submitted via a Pull Request (PR) and undergo review by another maintainer before merging.
```
Pull Request is in general written with first capital letters. Please check also other occurrences in the text and unify them.

File: images.md [hunk 10]
```diff
@@ -73,7 +68,6 @@ The following is a list of container images maintained in this organization. Mos
 | ------------ | ------------------- | -------------- |
 | [2.4](https://github.com/sclorg/httpd-container/tree/master/2.4) | [rhscl/httpd-24-rhel7](https://access.redhat.com/containers/#/registry.access.redhat.com/rhscl/httpd-24-rhel7) | [centos/httpd-24-centos7](https://hub.docker.com/r/centos/httpd-24-centos7/) |
 
-
 ## Thermostat
```
Comment (zmiklank) [inline images.md]: We do not really maintain this anymore, could be probably removed.

File: images.md [hunk 11]
```diff
@@ -87,7 +81,6 @@ The following is a list of container images maintained in this organization. Mos
 | ------------ | ------------------- | -------------- |
 | [s2i](https://github.com/sclorg/s2i-base-container.git) | [rhscl/s2i-base-rhel7](https://access.redhat.com/containers/#/registry.access.redhat.com/rhscl/s2i-base-rhel7) | [centos/s2i-base-centos7](https://hub.docker.com/r/centos/s2i-base-centos7/) |
 
-
 ## Developer Toolset
```
Comment (zmiklank) [inline images.md]: The same stands for developer toolset.

File: images.md [hunk 12]
```diff
@@ -97,15 +90,13 @@ The following is a list of container images maintained in this organization. Mos
...
-
 ## Redis
 
 | **version**  | **Red Hat Catalog** | **Docker Hub** |
 | ------------ | ------------------- | -------------- |
 | [3.2](https://github.com/sclorg/redis-container/tree/master/3.2) | [rhscl/redis-32-rhel7](https://access.redhat.com/containers/#/registry.access.redhat.com/rhscl/redis-32-rhel7) | [centos/redis-32-centos7](https://hub.docker.com/r/centos/redis-32-centos7/) |
 | [5](https://github.com/sclorg/redis-container/tree/master/5) | [rhscl/redis-5-rhel7](https://access.redhat.com/containers/#/registry.access.redhat.com/rhscl/redis-5-rhel7) |  |
 
-
 ## MongoDB
```
Comment (zmiklank) [inline images.md]: And also for MongoDB.
We support following container repositories:
httpd-container
nginx-container
s2i-php-container
varnish-container
mariadb-container
postgresql-container
s2i-nodejs-container
s2i-python-container
mysql-container
redis-container
s2i-perl-container
s2i-ruby-container

File: contribution.md [hunk 13]
```diff
@@ -1,117 +1,135 @@
...
-For RHEL, we install packages either from RHEL 7 base or from Red Hat [Software Collections](https://access.redhat.com/documentation/en/red-hat-software-collections/).
+For CentOS, we take packages from [SCLo SIG](http://wiki.centos.org/SpecialInterestGroup/SCLo). As a result, the images use RPM packages from the [Software Collections](http://softwarecollections.org).
 
+For RHEL, we install packages from RHEL 7 base or Red Hat [Software Collections](https://access.redhat.com/documentation/en/red-hat-software-collections/).
 
 ## Testing
 
-The images in this organization include test-suites, that test the images in basic use cases.
-The tests are being run on [Testing Farm](https://docs.testing-farm.io/general/0.1/index.html). Testing is done before the change is merged into the master branch. Therefore the place for triggering tests on demand is in a Pull Request.
+This organisation's images include test suites, which test the images in basic use cases.
```
Comment (zmiklank) [inline contribution.md]: I see occurrences of 'organisation' and 'organization' in this text. This should be united.

File: README.md [hunk 14]
```diff
@@ -1,58 +1,53 @@
-Welcome to the SCL.org Space
-============================
-This is basically an overview around the sclorg organization, aka what all one can find here.
+# Welcome to the SCL.org Space
```
Comment (zmiklank) [inline README.md]:
```suggestion
# Welcome to the Software Collections Organization
```

File: README.md [hunk 15]
```diff
@@ -1,58 +1,53 @@
...
 
-Why SCL.org? Since all the repositories, under this organization are somehow connected to the Software Collections concept. Software Collections give you the power to build, install, and use multiple versions of software on the same system, without affecting system-wide installed packages.
+This document offers a comprehensive introduction to the sclorg organization, highlighting the repositories under our umbrella and providing examples of their practical applications, along with links to relevant external resources.
 
-And since we have had set of new web development tools available as RPMs, tested and available in the directory on [softwarecollections.org](https://www.softwarecollections.org/en/), it is a good idea to re-use those tested packages and make container images from them, that are working fine in OpenShift.
+The primary focus of SCL.org revolves around the Software Collections concept. Software Collections offer a streamlined approach to building, installing, and utilizing multiple software versions on a single system, without disrupting pre-existing system-wide package installation.
```
Comment (zmiklank) [inline README.md]:
```suggestion
The primary focus of SCLOrg revolves around the Software Collections concept. Software Collections offer a streamlined approach to building, installing, and utilizing multiple software versions on a single system, without disrupting pre-existing system-wide package installation.
```

**Review by zmiklank — CHANGES_REQUESTED** [review]
Thank you for this PR. Please see my notes.

File: contribution.md [hunk 16]
```diff
@@ -1,117 +1,135 @@
-# Contribution to Container Images
-
-This document describes rules and best practices for contributing to http://github.com/sclorg/.
+# Contributing to Container Images
```
Comment (zmiklank) [inline contribution.md:1]: This is good opportunity to explain usage of "ready for review" label in contribution.md. Could you please state it somewhere here and also link a query for searching all open PRs in SCLOrg labeled with this label?

Comment (zmiklank) [reply contribution.md:1]: https://github.com/search?q=org%3Asclorg+label%3A%22ready+for+review%22+state%3Aopen&type=Issues

This is the query. Could you please double-check if it works as intended?

File: README.md (hunk 5 above)
Comment (zmiklank) [reply README.md]: Thanks for removing the "softwarecollections.org". However, I think that removing it causes the sentence to loose its meaning. I suggest removing whole sentence.

Comment (zmiklank) [reply README.md]: But wait a little please, I will rather ask Peto about this.

Comment (zmiklank) [reply README.md]: So the result is we really can remove it.
I saw one more occurrence of softwarecollections.org in README.md. Would you mind deleting all the occurrences please?

Comment (Lupphes) [reply README.md]: If it is confirmed, I can remove it. Thank you for the review. I will commit the changes.

**Review by zmiklank — APPROVED** [review]
Thank you for your work. LGTM.
